client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -62,9 +64,12 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	);
 
 	const urlQueryParams = useQuery();
+	const siteSlugOrIdInUrl = urlQueryParams.get( 'siteSlug' ) || urlQueryParams.get( 'siteId' );
+	const ref = urlQueryParams.get( 'ref' ) || '';
 
 	const site = useSite();
-	const ref = urlQueryParams.get( 'ref' ) || '';
+	const selectedSite = useSelector( ( state ) => site && getSite( state, site.ID ) );
+	const isFetchingSelectedSite = siteSlugOrIdInUrl && ! selectedSite;
```

Comment:
Is it possible to use the `isFetchingSite` selector? If the given site ID is wrong, the `isFetchingSelectedSite` will always be true since we cannot fetch the site successfully.

Reply:
Good idea, I used `isRequestingSite` to avoid the scenario you described.